ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/Metric.scala

```diff
@@ -141,13 +193,43 @@ object Metric {
   }
 
   object DelayMetric {
+
+    def empty[T](
+        recordTimeFunction: T => Seq[Timestamp],
+        objectives: List[DelayObjective],
+        clock: Clock,
+    ): DelayMetric[T] =
+      DelayMetric(
+        recordTimeFunction = recordTimeFunction,
+        clock = clock,
+        objectives = objectives.map(objective => objective -> None).toMap,
+      )
+
     final case class Value(meanDelaySeconds: Option[Long]) extends MetricValue {
       override def formatted: List[String] =
         List(s"mean delay: ${meanDelaySeconds.getOrElse("-")} [s]")
     }
 
-    def empty[T](recordTimeFunction: T => Seq[Timestamp], clock: Clock): DelayMetric[T] =
-      DelayMetric(recordTimeFunction, clock)
+    sealed trait DelayObjective extends ServiceLevelObjective[Value]
```

Comment:
Are we going to have another `DelayObjective` aside from `MaxDelay`?

Reply:
I don't expect any. The `DelayObjective` trait was added to set the `SeviceLevelObjective[T]` type parameter to a specific type but I'm not attached to this solution :)

Reply:
But you can specify the type as well in one step no?
`final case class MaxDelay(maxDelaySeconds: Long) extends ServiceLevelObjective[Value]`

Reply:
You're right. It's a relic of the past commits :) I'll change this.